diff -up ./CMakeLists.txt.0001 ./CMakeLists.txt
--- ./CMakeLists.txt.0001	2014-07-16 20:52:37.674055690 +0300
+++ ./CMakeLists.txt	2014-07-16 20:54:02.443706615 +0300
@@ -275,7 +275,7 @@ include_directories("."
     ${BULLET_INCLUDE_DIRS}
 )
 
-link_directories(${SDL2_LIBRARY_DIRS} ${Boost_LIBRARY_DIRS} ${MYGUI_LIB_DIR})
+link_directories(${SDL2_LIBRARY_DIRS} ${Boost_LIBRARY_DIRS} ${MYGUI_LIB_DIR} ${MYGUI_LIBRARY_DIRS})
 
 if(MYGUI_STATIC)
    add_definitions(-DMYGUI_STATIC)
