From 8e99c75c44009ee6bbb55819b0115e626d8a91c9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ralf=20Cors=C3=A9pius?= <corsepiu@fedoraproject.org>
Date: Tue, 18 Aug 2015 05:33:52 +0200
Subject: [PATCH 4/4] Re-add osgframerenderer.

---
 examples/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/examples/CMakeLists.txt b/examples/CMakeLists.txt
index eeecfe5..507aa8b 100644
--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -47,6 +47,7 @@ IF(DYNAMIC_OPENSCENEGRAPH)
     ADD_SUBDIRECTORY(osgfont)
     ADD_SUBDIRECTORY(osgforest)
     ADD_SUBDIRECTORY(osgfxbrowser)
+    ADD_SUBDIRECTORY(osgframerenderer)
     ADD_SUBDIRECTORY(osgoutline)
     ADD_SUBDIRECTORY(osggameoflife)
     ADD_SUBDIRECTORY(osggeometry)
-- 
2.4.3

