diff -Naur a/etc.BSD/os-release b/etc.BSD/os-release
--- a/etc.BSD/os-release	2012-05-17 10:21:08.000000000 +0600
+++ b/etc.BSD/os-release	2016-06-08 13:04:28.531303208 +0500
@@ -1,7 +1,7 @@
-NAME=Gentoo
-ID=gentoo
-PRETTY_NAME="Gentoo/BSD"
+NAME=Bentoo
+ID=bentoo
+PRETTY_NAME="Bentoo/BSD"
 ANSI_COLOR="1;32"
-HOME_URL="http://www.gentoo.org/"
-SUPPORT_URL="http://www.gentoo.org/main/en/support.xml"
-BUG_REPORT_URL="https://bugs.gentoo.org/"
+HOME_URL="https://bitbucket.org/redeyeteam/"
+SUPPORT_URL="https://bitbucket.org/redeyeteam/bentoo/wiki/"
+BUG_REPORT_URL="https://bitbucket.org/redeyeteam/bentoo/issues?status=new&status=open/"
diff -Naur a/etc.Linux/os-release b/etc.Linux/os-release
--- a/etc.Linux/os-release	2012-05-17 10:21:08.000000000 +0600
+++ b/etc.Linux/os-release	2016-06-08 11:23:04.000000000 +0500
@@ -1,7 +1,7 @@
-NAME=Gentoo
-ID=gentoo
-PRETTY_NAME="Gentoo/Linux"
+NAME=Bentoo
+ID=bentoo
+PRETTY_NAME="Bentoo/Linux"
 ANSI_COLOR="1;32"
-HOME_URL="http://www.gentoo.org/"
-SUPPORT_URL="http://www.gentoo.org/main/en/support.xml"
-BUG_REPORT_URL="https://bugs.gentoo.org/"
+HOME_URL="https://bitbucket.org/redeyeteam/"
+SUPPORT_URL="https://bitbucket.org/redeyeteam/bentoo/wiki/"
+BUG_REPORT_URL="https://bitbucket.org/redeyeteam/bentoo/issues?status=new&status=open/"
