diff -Naur a/install.rdf b/install.rdf
--- a/install.rdf	2013-08-15 13:49:06.000000000 +0600
+++ b/install.rdf	2015-11-07 09:56:05.464093678 +0500
@@ -1,11 +1,24 @@
 <?xml version='1.0' encoding='utf-8'?>
-<RDF:RDF xmlns:em="http://www.mozilla.org/2004/em-rdf#" xmlns:NC="http://home.netscape.com/NC-rdf#" xmlns:RDF="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
+<RDF xmlns="http://www.w3.org/1999/02/22-rdf-syntax-ns#" xmlns:em="http://www.mozilla.org/2004/em-rdf#">
+  <Description about="urn:mozilla:install-manifest">
+    <em:id>RightToClick@NettiCat</em:id>
+    <em:name>RightToClick</em:name>
+    <em:version>2.9.5.1-signed</em:version>
+    <em:type>2</em:type>
+    <em:description>Allows right clicks, highlighting and more where forbidden by javascript</em:description>
+    <em:creator>NettiCat</em:creator>
+    <em:optionsURL>"chrome://rtc/content/rtcOptions.xul"</em:optionsURL>
+    <em:homepageURL>"http://nc.ddns.us/extensions.html"</em:homepageURL>
+    <em:iconURL>"chrome://rtc/skin/rtclogo.png"></em:iconURL>
 
-  <RDF:Description RDF:about="rdf:#$ol9Cf" em:id="{ec8030f7-c20a-464f-9b0e-13a3a9e97384}" em:package="content/" em:locale="locale/en-US/" em:minVersion="4.0" em:maxVersion="23.*"/>
+    <!-- Firefox -->
+    <em:targetApplication>
+      <Description>
+        <em:id>{ec8030f7-c20a-464f-9b0e-13a3a9e97384}</em:id>
+        <em:minVersion>24.0</em:minVersion>
+        <em:maxVersion>35.0</em:maxVersion>
+      </Description>
+    </em:targetApplication>
 
-  <RDF:Description RDF:about="urn:mozilla:install-manifest" em:id="{cd617375-6743-4ee8-bac4-fbf10f35729e}" em:name="RightToClick" em:version="2.9.5.1-signed" em:type="2" em:description="Allows right clicks, highlighting and more where forbidden by javascript" em:creator="NettiCat" em:contributor="" em:optionsURL="chrome://rtc/content/rtcOptions.xul" em:homepageURL="http://nc.ddns.us/extensions.html" em:iconURL="chrome://rtc/skin/rtclogo.png">
-
-  <em:targetApplication RDF:resource="rdf:#$ol9Cf"/>
-  </RDF:Description>
-
-</RDF:RDF>
\ No newline at end of file
+    </Description>
+</RDF>
\ No newline at end of file
