diff -Naur lldb/cmake/modules/LLDBConfig.cmake b/cmake/modules/LLDBConfig.cmake
--- lldb/cmake/modules/LLDBConfig.cmake	2016-12-14 00:24:25.000000000 +0500
+++ b/cmake/modules/LLDBConfig.cmake	2017-03-19 11:30:57.743440519 +0500
@@ -182,6 +182,7 @@
       add_definitions( -DLLDB_PYTHON_HOME="${LLDB_PYTHON_HOME}" )
     endif()
   else()
+    set(Python_ADDITIONAL_VERSIONS 2.7)
     find_package(PythonLibs REQUIRED)
   endif()
   
diff -Naur lldb/cmake/modules/LLDBStandalone.cmake b/cmake/modules/LLDBStandalone.cmake
--- lldb/cmake/modules/LLDBStandalone.cmake	2016-12-14 00:24:25.000000000 +0500
+++ b/cmake/modules/LLDBStandalone.cmake	2017-03-19 11:30:57.743440519 +0500
@@ -74,7 +74,7 @@
   include(HandleLLVMOptions)
 
   if (PYTHON_EXECUTABLE STREQUAL "")
-    set(Python_ADDITIONAL_VERSIONS 3.5 3.4 3.3 3.2 3.1 3.0 2.7 2.6 2.5)
+    set(Python_ADDITIONAL_VERSIONS 2.7 3.5 3.4 3.3 3.2 3.1 3.0 2.6 2.5)
     include(FindPythonInterp)
     if( NOT PYTHONINTERP_FOUND )
       message(FATAL_ERROR
diff -Naur lldb/scripts/Python/modules/readline/readline.cpp b/scripts/Python/modules/readline/readline.cpp
--- lldb/scripts/Python/modules/readline/readline.cpp	2016-12-14 00:24:25.000000000 +0500
+++ b/scripts/Python/modules/readline/readline.cpp	2017-03-19 11:30:57.744440506 +0500
@@ -90,7 +90,7 @@
 #if PY_MAJOR_VERSION >= 3
     return PyModule_Create(&readline_module);
 #else
-    Py_InitModule4(
+    Py_InitModule4_64(
         "readline",
         moduleMethods,
         moduleDocumentation,
