--- utils/build-script-impl.old	2016-11-12 15:19:11.140145939 +0100
+++ utils/build-script-impl.old	2016-11-12 15:19:20.470145957 +0100
@@ -827,7 +827,7 @@ function configure_default_options() {
         local default_value="${KNOWN_SETTINGS[$((i+1))]}"
 
         # Find the variable name in our lookup table.
-        local varname="${known_setting_varnames[$((i/3))]]}"
+        local varname="${known_setting_varnames[$((i/3))]}"
 
         # Establish the associative array mapping.
         eval "${setting//-/_}_VARNAME=${varname}"
