diff -Naur a/install.rdf b/install.rdf
--- a/install.rdf	2013-02-26 17:14:56.000000000 +0600
+++ b/install.rdf	2015-11-07 09:44:44.345608831 +0500
@@ -1,8 +1,7 @@
 <?xml version='1.0' encoding='utf-8'?>
 <RDF xmlns="http://www.w3.org/1999/02/22-rdf-syntax-ns#" xmlns:em="http://www.mozilla.org/2004/em-rdf#">
-<Description about="urn:mozilla:install-manifest">
-    <em:id>{25A1388B-6B18-46c3-BEBA-A81915D0DE8F}</em:id>
-    <em:name>Quick Locale Switcher</em:name>
+  <Description about="urn:mozilla:install-manifest">
+    <em:id>Quick Locale Switcher@Martijn Kooij</em:id>
     <em:version>1.7.8.5.1-signed</em:version>
     <em:description>Quickly change the general.useragent.locale preference.</em:description>
     <em:creator>Martijn Kooij</em:creator>
@@ -11,82 +10,22 @@
     <em:iconURL>chrome://qls/skin/qls_32x32.png</em:iconURL>
     <em:type>2</em:type>
 
-    <em:contributor>Original author Martijn Kooij</em:contributor>
-    <em:contributor>Thanks to LXweb.nl for the GraphX</em:contributor>
-    <em:contributor>Thanks to Mark James of famfamfam.com (country flags)</em:contributor>
-    <em:contributor>Thanks to Rick of rick.jinlabs.com (gl-ES flag)</em:contributor>
-    <em:contributor>Thanks to babelzilla.org</em:contributor>
-    <em:contributor>-</em:contributor>
-    <em:contributor>Goofy (fr)</em:contributor>
-    <em:contributor>MatrixIsAllOver (it)</em:contributor>
-    <em:contributor>Chuzo (es-ES)</em:contributor>
-    <em:contributor>Victory (ja-JP until 1.4)</em:contributor>
-    <em:contributor>seabow (ja-JP until 1.5)</em:contributor>
-    <em:contributor>Premier (ja until 1.6.7)</em:contributor>
-    <em:contributor>Fenian (de-DE)</em:contributor>
-    <em:contributor>Batuhan Çetin (tr)</em:contributor>
-    <em:contributor>BlueRobot (ko-KR)</em:contributor>
-    <em:contributor>Teo (pl)</em:contributor>
-    <em:contributor>Petr Šimek (cs-CZ)</em:contributor>
-    <em:contributor>Emu (bg-BG)</em:contributor>
-    <em:contributor>Zefranc (pt-PT)</em:contributor>
-    <em:contributor>Miles (sl-SI)</em:contributor>
-    <em:contributor>Fang5566 (zh-CN)</em:contributor>
-    <em:contributor>Wladow (sk-SK)</em:contributor>
-    <em:contributor>TLemur (ru-RU)</em:contributor>
-    <em:contributor>moZes (fy-NL)</em:contributor>
-    <em:contributor>Winiah (zh-TW)</em:contributor>
-    <em:contributor>Teboga (pt-BR)</em:contributor>
-    <em:contributor>AlleyKat (da)</em:contributor>
-    <em:contributor>Garas (lt-LT)</em:contributor>   
-    <em:contributor>miz (ja-JP-mac) (until 1.5)</em:contributor>   
-    <em:contributor>Regueifeiro (gl-ES)</em:contributor>
-    <em:contributor>Besnik (sq-AL)</em:contributor>
-    <em:contributor>siiie (he-IL)</em:contributor>
-    <em:contributor>Sonickydon (el)</em:contributor>
-    <em:contributor>Ian Moody (en-GB)</em:contributor>
-    <em:contributor>Pedram Veisi (fa-IR)</em:contributor>
-    <em:contributor>phireak (km-KH)</em:contributor>
-    <em:contributor>Khoruzhin (uk-UA)</em:contributor>
-    <em:contributor>Krcko (hr-HR)</em:contributor>
-    <em:contributor>Andrew O. Shadoura (be-BY)</em:contributor>
-    <em:contributor>tonynguyen (vi-VN)</em:contributor>
-    <em:contributor>dan.lungescu (ro-RO)</em:contributor>
-    <em:contributor>acushnir (es-AR)</em:contributor>
-    <em:contributor>gLes (hu-HU)</em:contributor>
-    <em:contributor>gandhi2 (nb-NO)</em:contributor>
-    <em:contributor>bscd (fi)</em:contributor>
-    <em:contributor>Opakoman (ca-AD)</em:contributor>
-    <em:contributor>nassimjd (ar)</em:contributor>
-    <em:contributor>priyanhere (ml-IN)</em:contributor>
-    <em:contributor>srtxg (wa-BE)</em:contributor>
-    <em:contributor>kristjan (is-IS)</em:contributor>
-    <em:contributor>Ivan Jonoski (mk-MK)</em:contributor>
-    <em:contributor>Ekko (nn-NO)</em:contributor>
-    <em:contributor>ljushaff (sr-RS)</em:contributor>
-    <em:contributor>sorbzilla (eo)</em:contributor>
-    <em:contributor>sorbzilla (dsb)</em:contributor>
-    <em:contributor>sorbzilla (hsb)</em:contributor>
-    <em:contributor>Haebaru (ja-JP)</em:contributor>
-    <em:contributor>Lakrits (sv-SE)</em:contributor>
-    <em:contributor>akerbeltz (gd)</em:contributor>
-
-<!-- Mozilla Firefox -->
-<em:targetApplication>
-    <Description>
+    <!-- Firefox -->
+    <em:targetApplication>
+      <Description>
         <em:id>{ec8030f7-c20a-464f-9b0e-13a3a9e97384}</em:id>
-        <em:minVersion>4.0</em:minVersion>
-        <em:maxVersion>21.0a1</em:maxVersion>
-    </Description>
-</em:targetApplication>
-<!-- Mozilla Thunderbird -->
-<em:targetApplication>
-    <Description>
+        <em:minVersion>24.0</em:minVersion>
+        <em:maxVersion>35.0</em:maxVersion>
+      </Description>
+    </em:targetApplication>
+
+    <!-- Thunderbird -->
+    <em:targetApplication>
+      <Description>
         <em:id>{3550f703-e582-4d05-9a08-453d09bdfdc6}</em:id>
         <em:minVersion>5.0</em:minVersion>
         <em:maxVersion>21.0a1</em:maxVersion>
     </Description>
-</em:targetApplication>
-
-</Description>
+    </em:targetApplication>
+    </Description>
 </RDF>
\ No newline at end of file
